orchagent/crmorch.cpp

```diff
@@ -394,10 +395,6 @@ void CrmOrch::doTask(SelectableTimer &timer)
     getResAvailableCounters();
     updateCrmCountersTable();
     checkCrmThresholds();
-
-    auto interv = timespec { .tv_sec = m_pollingInterval.count(), .tv_nsec = 0 };
```

Comment:
Why are these lines removed?. This is currently in master as well.

Reply:
Good you asked, as I was unable to add comments for lines to be deleted. 

In short: redundant.

You need to set interval, only when it changes. The change is handled at CrmOrch::handleSetCommand.

This code CrmOrch::doTask(SelectableTimer &timer) is called, each time the timer is fired. No need to keep setting the same value, on each call.

If you look at other agents, like AclOrch::doTask(SelectableTimer &timer), they don't do that, as it is redundant.

Reply:
Yes, originally, set command didn't have it. It was reset during the next timer expire event. Now that we've it in set, we would need to fix this in master as well and later cherry-pick?

Reply:
m_pollingInterval can be modified. So this snippet is needed.

---
In reply to: [295486739](https://github.com/Azure/sonic-swss/pull/944#discussion_r295486739) [](ancestors = 295486739)

Reply:
CrmOrch::handleSetCommand handles any change to m_pollinginterval.

Reply:
Let me drop this change here and put it in masters, which can be cherry picked into Mar, as suggested by Prince.